### PR TITLE
HW 7: Task 1

### PR DESCRIPTION
**Pull request by ValeriL**

### CODE REVIEW

File: Homework/homework7/task1.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    count = 0
+    for el in collection:
+        count += el == element
+        if isinstance(el, (dict, tuple, list, set)):
+            count += count_occurrences(el, element)
+
+    return count
+
+
+def find_occurrences(tree: dict, element: Any) -> int:
```
Comment (Cjkjvfnby) [inline Homework/homework7/task1.py:32]: What the difference between `find_occurrences` and `count_occurrences`?

File: Homework/homework7/task1.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    str, list, tuple, dict, set, int, bool
+"""
+from itertools import chain
+from typing import Any, Iterable
+
+
+def chain_collection(collection: Iterable):
+    if isinstance(collection, dict):
+        return chain.from_iterable(collection.items())
+    return chain(collection)
```
Comment (Cjkjvfnby) [inline Homework/homework7/task1.py:16]: What `chain`, `dict` and other `collection` types have in common?

**Review by Cjkjvfnby — COMMENTED** [review]
This solution is overcomplicated but it works.  

Let's try to make it better, it will require some comment exchange, so it will require time.  That why  I marked it as done on my table.  Please back to this task only after the other tasks are submited.

File: Homework/homework7/task1.py (hunk 1 above)
Comment (ValeriL) [reply Homework/homework7/task1.py:32]: Actually, almost nothing. Only a type of first argument. For `find_occurrences` it should be only dictionary. Thats only why I created an extra function.

Comment (Cjkjvfnby) [reply Homework/homework7/task1.py:32]: And what are the relations between types of the first argument?

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**